### PR TITLE
Extract file selection Service and device selection

### PR DESCRIPTION
**Pull request by xnkevinnguyen**
# Description:
Refactoring task to make the code more readable. 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Limitations:

Please describe limitations of this PR

# Testing:

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] My code has been formatted with `npm run format` and passes the checks in `npm run check`
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/service/fileSelectionService.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    //         currentTextDocument &&
+    //         utils.getActiveEditorFromPath(currentTextDocument.fileName) ===
+    //             undefined
+    //     ) {
+    //         await vscode.window.showTextDocument(
+    //             currentTextDocument,
+    //             vscode.ViewColumn.One
+    //         );
+    //     }
+    // };
```
Comment (vandyliu) [inline src/service/fileSelectionService.ts]: We should remove this.

Comment (xnkevinnguyen) [reply src/service/fileSelectionService.ts]: Removed. Thanks for noticing

File: src/service/fileSelectionService.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    ): vscode.TextDocument => {
+        const activeEditor = vscode.window.visibleTextEditors.find(
+            (editor: vscode.TextEditor) => editor.document.fileName === filePath
+        );
+        return activeEditor ? activeEditor.document : undefined;
+    };
+    private getActivePythonFile = () => {
+        const editors: vscode.TextEditor[] = vscode.window.visibleTextEditors;
+        const activeEditor = editors.find(
+            editor => editor.document.languageId === "python"
```
Comment (isadorasophia) [inline src/service/fileSelectionService.ts]: Consider making "python" a constant for these identifiers? We already have one defined at `GLOBAL_ENV_VARS.PYTHON`, but maybe we should add a new one specifically for VS Code language IDs... Anyway, it's up to you!

File: src/extension.ts [hunk 3]
```diff
@@ -527,9 +511,13 @@ export async function activate(context: vscode.ExtensionContext) {
...
         } else {
             // Save on run
-            await currentTextDocument.save();
+            await fileSelectionService.getCurrentTextDocument().save();
 
-            if (!currentTextDocument.fileName.endsWith(".py")) {
+            if (
+                !fileSelectionService
+                    .getCurrentTextDocument()
+                    .fileName.endsWith(".py")
```
Comment (isadorasophia) [inline src/extension.ts]: We can also make ".py" a constant

**Review by isadorasophia — APPROVED** [review]
I have no idea how to comment on something that has not been edited by the PR <_<

While you are on it, it would be nice if we could also get rid of all these noisy `console.log` at `extension.ts` (I noticed you already removed some of them).

Otherwise, nice clean up! 🎡

File: src/service/fileSelectionService.ts (hunk 2 above)
Comment (xnkevinnguyen) [reply src/service/fileSelectionService.ts]: done!

**Pull request merged**